capstone-backend/index.js

```diff
@@ -46,32 +46,66 @@ app.get('/users', async (req, res) => {
 app.get('/home', (req, res) => {
   res.send('<h1>bruhhh</h1>')
 })
+let CanadianBorders = [];
+let MexicanBorders = [];
+let crossingData = [];
 
-let crossingData= [];
 const fetchCrossingData = async () => {
-  await fetch('https://bwt.cbp.gov/api/bwtrss/getAllPortsRss/Mexico')
+  await fetch('https://bwt.cbp.gov/xml/bwt.xml')
   .then(response => {
       if (!response.ok) {
         throw new Error(`HTTP error! status: ${response.status}`);
       }
       return response.text();
     })
     .then(data => {
+      //look into settings, to prevent to be displayed as array of only one item
       parser.parseString( data, (err, result) => {
         if (err) {
             console.error('error', err);
             return;
         }
-        crossingData = result.rss.channel[0].item;
+        crossingData = result;
+        console.log(crossingData)
     })
     })
     .catch(error => {
       console.error(`Error fetching data: `, error);
     });
+    for (const item in crossingData.border_wait_time.port) {
+      if (crossingData.border_wait_time.port[item].border == "Canadian Border") {
+        CanadianBorders.push({ 
+          border : crossingData.border_wait_time.port[item].border,
+          borderRegion : crossingData.border_wait_time.port[item].port_name,
+          crossingName : crossingData.border_wait_time.port[item].crossing_name,
+          hours : crossingData.border_wait_time.port[item].hours,
+          portStatus : crossingData.border_wait_time.port[item].port_status,
+          passengerVehicleWait : crossingData.border_wait_time.port[item].passenger_vehicle_lanes[0].standard_lanes[0].delay_minutes,
+          passengerVehicleWait : crossingData.border_wait_time.port[item].pedestrian_lanes[0].standard_lanes[0].delay_minutes
+      }) }
+      else {
+        MexicanBorders.push({ 
+          border : crossingData.border_wait_time.port[item].border,
+          borderRegion : crossingData.border_wait_time.port[item].port_name,
+          crossingName : crossingData.border_wait_time.port[item].crossing_name,
+          hours : crossingData.border_wait_time.port[item].hours,
+          portStatus : crossingData.border_wait_time.port[item].port_status,
+          passengerVehicleWait : crossingData.border_wait_time.port[item].passenger_vehicle_lanes[0].standard_lanes[0].delay_minutes,
+          pedestrianLaneWait : crossingData.border_wait_time.port[item].pedestrian_lanes[0].standard_lanes[0].delay_minutes
+      })
+      }
+    }
+    crossingData = { 
+    lastUpdatedDate : crossingData.border_wait_time.last_updated_date,
+    lastUpdatedTime : crossingData.border_wait_time.last_updated_time,
+    numOfPorts : crossingData.border_wait_time.number_of_ports,
+    allMexicanPorts : MexicanBorders,
+    allCanadianPorts : CanadianBorders
+};
 }
 
 app.get('/borderdata', (req, res) => {
-  fetchCrossingData();
+  fetchCrossingData()
```

Comment:
You can install Prettier extension on your VS code. Should help you format your file and add semi colons where missing